pkg/operator/reconciler/run/reconciler.go

```diff
@@ -248,6 +248,10 @@ func (r *Reconciler) uploadLogs(ctx context.Context, run *obj.Run, plr *obj.Pipe
 }
 
 func (r *Reconciler) uploadLog(ctx context.Context, namespace, podName, containerName string) (string, error) {
+	if r.StorageClient == nil {
+		return "", nil
```

Comment:
Perhaps if we return an empty string here we should avoid setting the `Logs` field for the step altogether?

Reply:
I hesitated to do that, as some of the legacy code in the API does not handle this very well. I believe I hardened that code  in the API already, but it had some fairly inflexible expectations about logging. The result of that was limited to throwing abnormal (and erroneous) error information into step logs in the UI, but it wasn't particularly ideal from a user perspective. That is probably no longer necessary, but that would need further evaluation on the API side, and this was aimed at status quo.

Reply:
And IIRC, what this currently does is support Pod log streaming to the UI while the step is running. If we removed the `Logs` field altogether it would shut off both, based off the legacy handling that's in place.

We should be removing all of this legacy functionality soon, so I'm not as prone to refactor this as I normally would be.